code/game/objects/items/storage/toolbox.dm

```diff
@@ -322,20 +311,21 @@
 	/// Tray we steal the og contents from.
 	var/obj/item/surgery_tray/tray_type = /obj/item/surgery_tray
 
-/obj/item/storage/toolbox/medical/Initialize(mapload)
-	. = ..()
-	// what do any of these numbers fucking mean
-	atom_storage.max_total_storage = 20
-	atom_storage.max_slots = 11
-
 /obj/item/storage/toolbox/medical/PopulateContents()
-	var/atom/fake_tray = new tray_type(get_turf(src)) // not in src lest it fill storage that we need for its tools later
-	for(var/atom/movable/thingy in fake_tray)
-		thingy.forceMove(src)
+	atom_storage.max_slots = 0
+	atom_storage.max_total_storage = 0
+
+	var/atom/fake_tray = new tray_type(null)
+	for(var/obj/item/tool in fake_tray)
+		tool.forceMove(src)
+		atom_storage.max_slots += 1
+		atom_storage.max_total_storage += tool.w_class
+
 	qdel(fake_tray)
 
 /obj/item/storage/toolbox/medical/full
 	tray_type = /obj/item/surgery_tray/full
+	stor
```

Comment:
?

Reply:
How did this compile????

Reply:
same as s// compiled for like, 8 years. Relative typepaths.